packages/shopping/src/__tests__/acceptance/user-management.controller.acceptance.ts

```diff
@@ -39,9 +38,11 @@ describe('UserController', () => {
   };
 
   const userPassword = 'p4ssw0rd';
+  const jwtSecret = 'a-secret-that-is-at-least-32-bytes-long';
   let expiredToken: string;
 
   before('setupApplication', async () => {
+    process.env.JWT_SECRET = jwtSecret;
```

Comment:
Why is this necessary? ideally, nothing should break on the tests side based on the change you did since it was "backward" compatible

Reply:
It was backward compatible on my initial proposal, not after changing to have a random secret to be generated as default.
In the tests there is an expired key generated. It has to be generated with the same key as used by the application.
If random - it is unknown. Need to change tests to have a fixed known key.
Also, it is no longer fully backwards compatible; if having multiple instances and not setting JWT_SECRET, each instance will use a different secret; a key generated with one instance using random secret will not validate on other instances as they will have a different key.

Reply:
Okay will have a look once more. This approach feels hacky! Thanks for highlighting this

Reply:
The PR looks good now in my opinion. well done @johanschvn

@mrmodise I don't think the implementation is hacky, just addressing the points mentioned:

> If random - it is unknown. Need to change tests to have a fixed known secret.

That's a common approach to use a static/hardcoded secret during testing.

> Also, it is no longer fully backwards compatible; if having multiple instances and not setting JWT_SECRET, each instance will use a different secret; a secret generated with one instance using random secret will not validate on other instances as they will have a different secrets.

That is probably not a problem, I rather see it as a problem if someone uses the hardcoded secret of the module. And it makes sense to advocate good security practices in the example app as developers will use it as a starting point to build their own application which might be used in production in the end.

Reply:
@hacksparrow and @jannyHou your thoughts? I just think this would be more ideal solved at the security service model than to have multiple, probably, different implementations to achieve essentially the same thing.

I understand the common pattern of generating hard coded values on tests but wouldn't it rather paint a better picture from a testing perspective if we used a sample generated string from the crypo module?

Reply:
> I just feel passing the hard coded string to an environment variable is unnecessary since ideally this should be managed outside of the application hence my thoughts on this being hacky. I mean this approach defeats the intended cause of good security practices

OK, I will try to change the test to instead read the randomly generated JWT secret from app context and use that for generating the expired token.

Reply:
Cool

Reply:
@mrmodise I have changed the test script now so it does not change/set environment JWT_SECRET; I still have to change the generation of the expired token; both the app and test code generating an expired token must use the same secret. I hope @nflaig likes this change too!?